packages/plugin/src/Fields/Implementations/ValueGenerators/AssetSourceGenerator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Freeform for Craft CMS.
+ *
+ * @author        Solspace, Inc.
+ * @copyright     Copyright (c) 2008-2022, Solspace, Inc.
+ *
+ * @see           https://docs.solspace.com/craft/freeform
+ *
+ * @license       https://docs.solspace.com/license-agreement
+ */
+
+namespace Solspace\Freeform\Fields\Implementations\ValueGenerators;
+
+use Solspace\Freeform\Attributes\Property\Property;
+use Solspace\Freeform\Attributes\Property\ValueGeneratorInterface;
+
+class AssetSourceGenerator implements ValueGeneratorInterface
+{
+    public function generateValue(Property $property, string $class, ?object $referenceObject): ?int
+    {
+        $volumes = \Craft::$app->getVolumes()->getViewableVolumes();
+
+        if (\count($volumes)) {
+            return $volumes[0]->id;
+        }
+
+        return null;
```

Comment:
Probably can be written as something like this..

```suggestion
        return $volumes[0]->id ?? null;
```